src/compiler/emitter.ts

```diff
@@ -3925,6 +3925,12 @@ module ts {
                     write(", ");
                     emit(imp.name);
                 });
+                forEach(node.amdDependencies, amdDependency => {

```

Comment:
what if you have multiple dependencies, ones with names and ones without. this will get the order our of sync, since name is optional and path is not. e.g.:
```TypeScript
///<amd-dependency path='module1' />
///<amd-dependency path='module2' name='m2'/>
```

will emit:
```JavaScript
define(["require", "exports", "module1", "module2"], function (require, exports, m2) { }
```

which is not what you want. so either you reorder them, so that ones with names come first, which we do not do usually as we like to keep emitted code as close as possible to generated code. or you give the other ones unused random names consider using createTempVariable utility in the emitter.

Also worth adding a new test for this case.